dvc/tree/gdrive.py

```diff
@@ -510,22 +541,33 @@ def _path_to_item_ids(self, path, create, use_cache):
         )
 
     def _get_item_id(self, path_info, create=False, use_cache=True, hint=None):
-        assert path_info.bucket == self._bucket
+        if self.path_info:
+            assert path_info.bucket == self._bucket
+        else:
+            # cache is not used when Tree is associated with a GDriveDependency
+            use_cache = False
 
-        item_ids = self._path_to_item_ids(path_info.path, create, use_cache)
+        item_ids = self._path_to_item_ids(
+            path_info.path, create, use_cache, path_info.bucket
+        )
         if item_ids:
-            return min(item_ids)
+            item_id = min(item_ids)
+            if item_id == path_info.bucket:
+                # if item_id doesn't exist, _fetch_metadata
+                # raises FileMissingError
+                self._fetch_metadata(item_id)
```

Comment:
minor: fetch only specific fields?

Reply:
Adding one filed might improve the performance. Solved!